src/external/BooksApiClient.ts

```diff
@@ -0,0 +1,36 @@
+import axios, { AxiosInstance } from 'axios';
+import { BooksResponse, SingleBookResponse } from '../models/Book';
+
+const baseUrl = "https://booksontap.azurewebsites.net/api";
+const booksEndpoint = baseUrl + "/books";
+
+const session: AxiosInstance = axios.create({
+    baseURL: baseUrl
+});
+
+export async function getAllBooks() {
+    let response = await session.get<BooksResponse>(booksEndpoint).then(function (response) {
+        // handle success
+        return response.data;
+        })
+        .catch(function (error) {
+        // handle error
+
+        console.log(error);
+        })
+    
+    return response;
+}
```

Comment:
You are mixing Promises with async/await. They work together, but async/await came along to make the pattern more readable and not have to chain `.then`'s.

```
try {
   let response = await session.get<BooksResponse>(booksEndpoint)
   let books = response.data

   return books
} catch (err) {
   console.log(error)
}


```